praw/objects.py

```diff
@@ -1899,6 +1902,34 @@ def remove_editor(self, username, *args, **kwargs):
         return self.add_editor(username=username, _delete=True, *args,
                                **kwargs)
 
+    def revisions(self, **kwargs):
+        """Return a dictionary of all page revisions.
+
+        Additional parameters are passed to
+        :meth:`~praw.__init__.BaseReddit.request_json` to be used as
+        URL parameters.
+        """
+        url = self.reddit_session.config['wiki_page_revisions'].format(
+                subreddit=six.text_type(self.subreddit),
+                page=self.page.lower())
+
+        prev_use_oauth = self.reddit_session._use_oauth
+        self.reddit_session._use_oauth = True
+        response = self.reddit_session.request_json(url, params=kwargs)
+        self.reddit_session._use_oauth = prev_use_oauth
+
+        return response['data']
+
+    def revision(self, version):
+        """Return a specific version of the WikiPage.
+
+        :param version: The version GUID of the page (returned by
+        :meth:`revisions`)
+        """
+        result = WikiPage(self.reddit_session, self.subreddit, self.page,
```

Comment:
Question about wiki revisions since I just don't work with them often enough, does what's returned in `self.revisions()` contain a dictionary of UUIDS to objects, or something else. If it's a dictionary of uuids to objects, it would be nice to cache those objects internally (self._revisions_cache in a dictionary to trade speed for memory usage, since memory usage would be minimal.